### PR TITLE
Fix: Windows scoped package configs (fixes #5644)

### LINKED ISSUE #5644: Scoped npm packages on Windows do not seem to work
Hello,

it looks like scoped packages do not work correctly on Windows machines. Below you can see that eslint is trying to load a configuration from `@strv\javascript\standard\optional.js/eslint-config`, while the correct place to load it from would be `@strv\eslint-config-javascript\standard\optional.js`.

To provide more context, the file `@strv/javascript/environments/nodejs/optional.js` contains a **relative** reference to another configuration file via `extends` property:
```js
module.exports = {
  extends: '../../standard/optional.js'
}
```

If it helps, the whole configuration package is publicly available [here](https://github.com/strvcom/js-coding-standards).

I wonder if this is an issue with ESLint or with the configuration package. On OS X, this configuration package works without issues.

Thank you for your help!

**What version of ESLint are you using?** 2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```js
// .eslintrc.js
module.exports = {
  extends: [
    '@strv/javascript/environments/nodejs/latest',
    '@strv/javascript/environments/nodejs/best-practices',
    '@strv/javascript/environments/nodejs/optional'
  ],

  rules: {
    'valid-jsdoc': 2
  }
}
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
Cannot find module '@strv\javascript\standard\optional.js/eslint-config' from 'C:\project-root\node_modules\@strv\eslint-config-javascript\environments\nodejs'
Referenced from: @strv/javascript/environments/nodejs/optional
Referenced from: C:\project-root\.eslintrc.js
Error: Cannot find module '@strv\javascript\standard\optional.js/eslint-config' from 'C:\project-root\node_modules\@strv\eslint-config-javascript\environments\nodejs'
Referenced from: @strv/javascript/environments/nodejs/optional
Referenced from: C:\project-root\.eslintrc.js
   at Function.module.exports [as sync] (C:\project-root\node_modules\resolve\lib\sync.js:33:11)
   at resolve (C:\project-root\node_modules\eslint\lib\config\config-file.js:416:38)
   at load (C:\project-root\node_modules\eslint\lib\config\config-file.js:435:24)
   at C:\project-root\node_modules\eslint\lib\config\config-file.js:355:36
   at Array.reduceRight (native)
   at applyExtends (C:\project-root\node_modules\eslint\lib\config\config-file.js:338:28)
   at load (C:\project-root\node_modules\eslint\lib\config\config-file.js:462:22)
   at C:\project-root\node_modules\eslint\lib\config\config-file.js:355:36
   at Array.reduceRight (native)
   at applyExtends (C:\project-root\node_modules\eslint\lib\config\config-file.js:338:28)
```

/cc @MartinStava @dostalleos

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

File: lib/config/config-file.js [hunk 1]
```diff
@@ -513,6 +529,7 @@ module.exports = {
     resolve: resolve,
     write: write,
     applyExtends: applyExtends,
+    normalizePackageName: normalizePackageName,
```
Comment (nzakas) [inline lib/config/config-file.js]: Made this public so I could write tests for it.

File: lib/config/config-file.js [hunk 2]
```diff
@@ -392,6 +392,16 @@ function applyExtends(config, filePath, relativeTo) {
...
  */
 function normalizePackageName(name, prefix) {
+
+    /*
+     * On Windows, name can come in with Windows slashes instead of Unix slashes.
+     * Normalize to Unix first to avoid errors later on.
+     * https://github.com/eslint/eslint/issues/5644
+     */
+    if (name.indexOf("\\") > -1) {
+        name = name.replace(/\\/g, "/");
```
Comment (gyandeeps) [inline lib/config/config-file.js]: We already have function exposed in utils doing similar stuff: https://github.com/eslint/eslint/blob/master/lib/util/path-util.js#L26
I think we should use that function here.

Comment (nzakas) [inline lib/config/config-file.js]: Good call.

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**